components/calendar/src/calendar_arithmetic.rs

```diff
@@ -132,4 +142,35 @@ impl<C: CalendarArithmetic> ArithmeticDate<C> {
     pub fn day_of_month(&self) -> types::DayOfMonth {
         types::DayOfMonth(self.day.into())
     }
+
+    /// The [`types::Month`] for the current month (with month code) for a solar calendar
+    /// Lunar calendars should not use this method and instead manually implement a month code
+    /// resolver.
+    ///
+    /// Returns "und" if run with months that are out of bounds for the current
+    /// calendar.
+    #[inline]
+    pub fn solar_month(&self) -> types::Month {
+        let code = match self.month {
+            a if a > C::months_for_every_year() => tinystr!(8, "und"),
+            1 => tinystr!(8, "M01"),
+            2 => tinystr!(8, "M02"),
+            3 => tinystr!(8, "M03"),
+            4 => tinystr!(8, "M04"),
+            5 => tinystr!(8, "M05"),
+            6 => tinystr!(8, "M06"),
+            7 => tinystr!(8, "M07"),
+            8 => tinystr!(8, "M08"),
+            9 => tinystr!(8, "M09"),
+            10 => tinystr!(8, "M10"),
+            11 => tinystr!(8, "M11"),
+            12 => tinystr!(8, "M12"),
+            13 => tinystr!(8, "M13"),
+            _ => tinystr!(8, "und"),
+        };
+        types::Month {
+            ordinal_month: self.month as u32,
+            code: types::MonthCode(code),
```

Comment:
Discussion: Since month codes are a well defined and stable syntax, should we define `MonthCode` to be stricter and have internal invariants, more like `Language` and `Region` subtags?

Reply:
They're not quite stable, more can be added, I didn't want to spec out a complete and future proof syntax

Reply:
ok. Once @ryzokuken writes the calendar spec, maybe we can strengthen the guarantee (for 2.0).